### PR TITLE
vmalert: support logs suppressing during config reloads

### PR DESCRIPTION
**Pull request by hagen1778**
The change is mostly required for ENT version of vmalert, since it supports object-storage for config files. Reading data from object storage could be time-consuming, so vmalert emits logs to track the progress.

However, these logs are mostly needed on start or on manual config reload. Printing these logs each time `rule.configCheckInterval` is triggered would too verbose. So the change allows to control logs emitting during config reloads.

Now, logs are emitted during start up or when SIGHUP is receieved. For periodicall config checks logs emitted by config pkg are suppressed.

### CODE REVIEW

File: app/vmalert/config/log/logger.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// but can be disabled via Disable method
+type Logger struct {
+	disabled bool
+}
+
+// Disable whether to ignore message logging.
+// Once disabled, logging continues to be ignored
+// until logger is enabled again.
+// Not thread-safe.
+func (l *Logger) Disable(v bool) {
```
Comment (zekker6) [inline app/vmalert/config/log/logger.go:15]: Wdyt about renaming this to `Supress` or maybe using 2 methods like disable/enable to make it more verbose?
Current naming is a bit confusing when reading:
```
cLogger.Disable(true)
defer cLogger.Disable(false)
```

Comment (hagen1778) [reply app/vmalert/config/log/logger.go]: updated

File: app/vmalert/config/log/logger.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package log
+
+import "github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+
+// Logger is using lib/logger for logging
+// but can be disabled via Disable method
+type Logger struct {
+	disabled bool
```
[the remark this answers is not in the document]
Comment (hagen1778) [reply app/vmalert/config/log/logger.go:13]: wanted to keep it simple. Updated

**Review by zekker6 — APPROVED** [review]
LGTM

**Comment by hagen1778** [comment]
@f41gh7 care to take another look?

**Comment by f41gh7** [comment]
Thanks!

**Pull request merged**